app.py

```diff
@@ -0,0 +1,61 @@
+import json
+from functools import wraps
+
+from flask import Flask, jsonify, abort, request
+
+def create_app():
+
+    settings = {
+        'DEBUG': True,
+    }
+
+    app = Flask(__name__)
+    app.config.update(settings)
+
+    return app
+
+app = create_app()
+
+
+def get_books():
+    books = {}
+
+    return books
+
+
+def save_book(book):
+    books = {}
+    return
+
+@app.route('/')
+def hello():
+    return 'Hi Mom!'
+
+@app.route('/books', methods=['GET'])
+def list_books():
+    books = get_books()
+    return jsonify(dict(books=books))
+
+
+@app.route('/books/<int:id>', methods=['GET'])
+def get_book(id):
+    books = get_books()
+    return abort(404)
+
+
+@app.route('/books', methods=['POST'])
+def add_book():
+    posted = request.get_json()
+    books_len = len(get_books())
+    book = dict(
```

Comment:
Again, you might want to rethink these routes in terms of events. 

I'd say the minimum we need is a GET `def list_events()` and a POST `def add_event()`. You can use a lot of what you already have here for that.